### DISCUSSION TITLE
A helpful eslint rule to require radashi namespace import with underscore

### DISCUSSION THREAD

**Discussion by aeharding** [radashi-org/radashi#290] [Announcements]
Hi there!

I am experimenting with Radashi and I like it so far.

One of my initial problems was a few of the functions have really generic names that often clash with variables I define - for example, `title()`.

For this reason, I decided that instead of

```ts
import { title } from "radashi";
```

I will do

```ts
import * as _ from "radashi";
```

However IDE import completion often prefers the former approach. So, I tried to get an eslint rule for it! The rule is defined as the following (credit to https://github.com/eslint/eslint/issues/19105#issuecomment-2458833155):

```js
"no-restricted-syntax": { 
  selector: "ImportDeclaration[source.value='radashi'][specifiers.length=0], ImportDeclaration[source.value='radashi'] > :matches(ImportSpecifier, ImportDefaultSpecifier)",
  message: "Use only `import * as _ from 'radashi'`"
} 
```

Result:

```ts
// allow only this:
import * as _ from "radashi"; // ✅ all good!

import { sort } from "radashi"; // ❌ not allowed
import "radashi"; // ❌ not allowed
import $ from "radashi"; // ❌ not allowed
```

-------

P.S. If this is a preferred pattern for Radashi, it would be possible to make a simple export with this rule in the Radashi project. With eslint v9, the devex would look something like this:

```ts
// eslint.config.js
import radashi from "radashi/eslint";

export default [
  ...radashi.recommended
 // rest of eslint config
]
```

**Comment by MarlonPassos-git** [comment]
I’m not too keen on making this a standard—perhaps more like a suggestion. For example, I personally always destructure functions. 

Also, this is something specific to ESLint, and in this project, we’re using Biome.js. 

Since it can be done with just two lines of code, maybe adding a page in the project documentation could be a simpler approach. If a lot of people show interest in this pattern, we could consider creating a package for ESLint or Biome.js in the future.

**Reply by aeharding** [reply]
Makes sense! This Github Discussion might just suffice too, it should be pretty easy to find.

**Reply by mctrafik** [reply]
In our org we definitely prefer the named imports because those treeshake better. 

But there's also an ability to create custom biome plugins now, so you can achieve that there, too, but I would not want to see this be "the standard".
